### PR TITLE
doc: Add a first pass contributing doc

### LINKED ISSUE #14: Create a `CONTRIBUTING.md`
# 📚 Documentation Issue Report

## Describe the issue

There are no contributing guidelines right now, and getting started is not straightforward. Especially not for devs who may be new to Go. 

## Describe the solution you'd like to see

We need clear guidelines for how you set up your dev environment, write code, and submit it. GitHub has [a guide](https://opensource.guide/starting-a-project/#writing-your-contributing-guidelines) for putting together a contributing doc, with links out to other guides.

[This](https://tartufo.readthedocs.io/en/stable/CONTRIBUTING.html) was one I wrote up previously, though it would need some massaging, I'm sure, for this project.

There are probably other great examples out there we can pull from as well.

### PR DESCRIPTION
**Pull request by tarkatronic**
Fixes #14

This is just a first pass at a contributing doc. It's based on ones from a few projects I've built in the past. It should cover pretty much everything... or at least be a good start.

### CODE REVIEW

**Comment by tarkatronic** [comment]
Note to self: I need to add mention in here that we require commits to be signed.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+If you've written Go before, you should be pretty familiar with our styling. All
+of our code is formatted with `gofmt`! If you have not written Go before, or
+would like a refresher, we highly encourage going through [Effective Go].
+
+We ask that you also enable [EditorConfig] support in your editor, to help in
+keeping things like spacing and line endings consistent.
+
+### Committing Code
+
+We generally follow [Conventional Commit] styling for our commit messages. While
```
Comment (violatido) [inline CONTRIBUTING.md:106]: [suggestion] You have so many helpful links for guiding folks to an organized approach of contributing! Perhaps you would like to include a link to a guide for Conventional Commits? For instance, [here](https://www.conventionalcommits.org/en/v1.0.0/#summary) is a breakdown from the conventional commits docs

**Review by violatido — APPROVED** [review]
Great writeup, Joey! I made a very minor suggestion, but as a newbie to Go and a person with limited open source development experience, this is a very helpful README that makes contribution much more approachable

File: CONTRIBUTING.md (hunk 1 above)
Comment (violatido) [reply CONTRIBUTING.md:106]: I was wondering about that! I haven't heard of referential markdown, but I definitely like that style better. Much easier to read!

**Pull request merged**